### ISSUE TITLE
Data explorer does not render the timestamp datatype 

### ISSUE THREAD

**Issue by trygu** [InseeFrLab/onyxia#690]
I've tried the Data explorer on a bunch of test-files, and it looks great. There is a small issue related to the rendering of timestamp  (the native date time datatype in Parequet). It is rendered as a numeric value, and not a date time. I'm attaching a test-file (parquet), and screenshots from both the data explorer and duckdb.

![Data explorer screenshot](https://github.com/InseeFrLab/onyxia/assets/3669823/3eb50761-2928-4f81-ac6f-7e0e4e48a3e3)
![Duckdb screenshot](https://github.com/InseeFrLab/onyxia/assets/3669823/37a33552-855d-41df-9c89-6bc3f6291938)

[output.parquet.zip](https://github.com/InseeFrLab/onyxia/files/13579358/output.parquet.zip)

**Comment by garronej** [comment]
Hey @trygu,  

Thanks for the feedback!  

Seems like an easy enough fix!

**Comment by RLesur** [comment]
@garronej I don't want to dampen your spirits, but it's not that simple. Clients (pyarrow, R arrow, Spark...) manage datetime differently. So, the content of a parquet file will depend on the client used to create it. This is a problem that @pengfei99 had documented [here](https://github.com/pengfei99/ParquetDataFormat#252-timestamp). Don't you remember, Jo? 😉

**Comment by trygu** [comment]
I think the problem is how you should represent a date time field; My suggestion is to represent it in a fixed iso-8601 fashion just like DuckDb. (and maybe at a later juncture, as a configurable option for the date time formatting).

**Comment by pengfei99** [comment]
@garronej The parquet format is the easiest one to work with. For the CSV format, you will have more trouble. For example, if a file is encoded with `window-1252`, and you try to open it with `UTF-8`, all the special characters will be wrongly interpreted. You will likely encounter this issue eventually.

**Comment by trygu** [comment]
> @garronej The parquet format is the easiest one to work with. For the CSV format, you will have more trouble. For example, if a file is encoded with `window-1252`, and you try to open it with `UTF-8`, all the special characters will be wrongly interpreted. You will likely encounter this issue eventually.

I agree, and I'm not sure if it's that important to fix this for CSV's, as you say they can be encoded (and not encoded) in a lot of different ways, but seldom as a large integer.

I think the most pressing issue is that the date time-type in Parquet is displayed using EPOCH. There are several strategies to GUESS that it is a date time-value, but the absolute best would be to get this information from the underlying reading of the metadata (of the parquet file), since the grid itself do not have this information. I also noticed that there is a milestone related to the File explorer. I would love this issue to be part of that milestone as well (nudge @fcomte).

**Comment by fcomte** [comment]
We made many improvement on this ? is it ok now @trygu ? @ddecrulle ?

**Comment by trygu** [comment]
Not yet! Its rendering now but it doesn't calculate the correct datetime (not using epoch?): ![image](https://github.com/user-attachments/assets/a5d7278e-5ccb-474a-8d8d-27bbe5767ab7)

The "uke" column is week and the datetime should be like the picture in the DuckDB example. Easely verifiable by testing the file I attached to this case.

**Comment by fcomte** [comment]
@ddecrulle can you look at this file s3://tm8enk/diffusion/output.parquet ?

 I think this is a millisecond timestamp but onyxia detects it as a microsecond timestamp as written in our explorer
